### PR TITLE
Introduce base for hosts listing page

### PR DESCRIPTION
**Pull request by jtomasek**
* Creates metalkube extension directory structure based on kubevirt extension
* Adds Hosts link to kubevirt nav extension
* Adds Metalkube flag and CRD in features
* Adds BaremetalHostModel
* Adds Basic HostsPage based on kubevirt vm listing

### CODE REVIEW

File: frontend/public/components/resource-pages.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { ReportReference, ReportGenerationQueryReference } from './chargeback';
 import { referenceForModel, GroupVersionKind } from '../module/k8s';
 import {
   AlertmanagerModel,
+  BaremetalHostModel,
```
[the remark this answers is not in the document]
Comment (jtomasek) [reply frontend/public/components/resource-pages.ts]: Done

File: frontend/public/metalkube/models/host.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+// TODO: reuse following from kubevirt-web-ui-components to avoid duplicity
+
+export const BaremetalHostModel: K8sKind = {
+  label: 'Host',
+  labelPlural: 'Hosts',
+  apiVersion: 'v1alpha1',
+  path: 'baremetalhosts',
+  apiGroup: 'metalkube.org',
+  plural: 'hosts',
```
Comment (honza) [inline frontend/public/metalkube/models/host.ts]: This should read:

```js
plural: 'baremetalhosts',
```

File: frontend/public/kubevirt/components/nav.jsx [hunk 3]
```diff
@@ -78,6 +78,7 @@ const PageNav = ({ onNavSelect, ResourceClusterLink, HrefLink, ResourceNSLink, M
       <NavSection title="Administration">
         <ResourceClusterLink resource="namespaces" name="Namespaces" required={FLAGS.CAN_LIST_NS} />
         <ResourceClusterLink resource="nodes" name="Nodes" required={FLAGS.CAN_LIST_NODE} />
+        <ResourceClusterLink resource="hosts" name="Hosts" required={FLAGS.METALKUBE} />
```
Comment (honza) [inline frontend/public/kubevirt/components/nav.jsx]: This should read:

```jsx
<ResourceClusterLink resource"baremetalhosts" name="Hosts" require={FLAGS.METALKUBE} />
```

Comment (jtomasek) [reply frontend/public/kubevirt/components/nav.jsx]: Done, thanks for the catch!

File: frontend/public/features.ts [hunk 4]
```diff
@@ -66,6 +68,7 @@ export enum FLAGS {
   CLUSTER_API = 'CLUSTER_API',
   CLUSTER_VERSION = 'CLUSTER_VERSION',
   MACHINE_CONFIG = 'MACHINE_CONFIG',
+  METALKUBE = 'METALKUBE',
```
[the remark this answers is not in the document]
Comment (jtomasek) [reply frontend/public/features.ts]: done

File: frontend/public/components/resource-pages.ts [hunk 5]
```diff
@@ -170,4 +171,5 @@ export const resourceListPages = ImmutableMap<GroupVersionKind | string, () => P
   .set(referenceForModel(PackageManifestModel), () => import('./operator-lifecycle-manager/package-manifest' /* webpackChunkName: "package-manifest" */).then(m => m.PackageManifestsPage))
   .set(referenceForModel(SubscriptionModel), () => import('./operator-lifecycle-manager/subscription' /* webpackChunkName: "subscription" */).then(m => m.SubscriptionsPage))
   .set(referenceForModel(InstallPlanModel), () => import('./operator-lifecycle-manager/install-plan' /* webpackChunkName: "install-plan" */).then(m => m.InstallPlansPage))
-  .set(referenceForModel(ClusterOperatorModel), () => import('./cluster-settings/cluster-operator' /* webpackChunkName: "cluster-operator" */).then(m => m.ClusterOperatorPage));
+  .set(referenceForModel(ClusterOperatorModel), () => import('./cluster-settings/cluster-operator' /* webpackChunkName: "cluster-operator" */).then(m => m.ClusterOperatorPage))
+  .set(referenceForModel(BaremetalHostModel), () => import('../metalkube/components/host/host' /* webpackChunkName: "baremetal-hosts" */).then(m => m.BaremetalHostsPage));
```
[the remark this answers is not in the document]
Comment (jtomasek) [reply frontend/public/components/resource-pages.ts]: done

**Pull request merged**